src/main/java/duke/commands/Command.java

```diff
@@ -0,0 +1,162 @@
+package duke.commands;
+
+import duke.tasks.*;
+import duke.storage.Storage;
+import duke.ui.Ui;
+import duke.exception.DukeException;
+
+public class Command {
+    public enum Commands {
+        TASK("task"), TODO("todo"), EVENT("event"), DEADLINE("deadline"), LIST("list"), BYE("bye");
+
+        private String command;
+
+        Commands(String command) {
+            this.command = command;
+        }
+
+        public String getCommand() {
+            return this.command;
+        }
+    }
+
+    private String[] elements;
+    private String commandWord;
+    private String fullInput;
+    private boolean isExit = false;
+
+    public Command(String fullInput) {
+        this.elements = fullInput.split(" ");
+        this.commandWord = elements[0];
+        this.fullInput = fullInput;
+    }
+
+    public boolean isExit() {
+        return this.isExit;
+    }
+
+    public void execute(TaskList taskList, Storage storage, Ui ui) throws DukeException {
+        switch (this.commandWord) {
+        case "bye":
+            ui.showMessage("Aww okay, see you next time!");
+            this.isExit = true;
+            break;
+        case "list":
+            if (taskList.getTaskList().size() < 1) {
```

Comment:
Maybe the task list and size of the task list could be placed outside of the switch statement as variables, to prevent extra calls to the `getTaskList()` and `size()` function